docs/Lua API.rst

```diff
@@ -2984,6 +2984,66 @@ Predefined instance methods:
 
 To avoid confusion, these methods cannot be redefined.
 
+customRawData
+=============
+
+A utility allowing for custom tags to be added to the raw definitions for modding purposes:
+
+* ``customRawData.getTag(typeDefinition, tag[, ...])``
+
+  Where ``typeDefinition`` is a raw struct as seen in ``df.global.world.raws``, e.g:
+  ``dfhack.gui.getSelectedItem().subtype``, ``tag`` is the name of the custom tag you
+  want to get, and the remaining arguments correspond to the arguments in the tag as
+  booleans denoting whether the argument in the tag is a string or a number. ``true``
+  for string, false (or absent) for number. It will then be returned using single or
+  multiple return values. If the tag is not present, the result is false, if it is
+  present but has no arguments, the result is true.
+
+* ``customRawData.getRaceCasteTag(raceDefinition, casteNumber, tag[, ...])``
+
+  Gets tag from a race or caste instead if appplicable. Everything with getTag applies to this.
```

Comment:
what is the type of `raceDefinition`? why do we need a separate method for race caste tags? Can this be handled by `getTag()`?

Reply:
No, it's separate behaviour. Or would you rather the caste business be merged in with if statements?